rfp-proposals/filexplorer.org.md

```diff
@@ -0,0 +1,180 @@
+4. Work-in-progress staging site deployed with:
+	1. Working basic data retrieval via back-end and MongoDB.
+	2. Fully implemented, responsive, header, footer, and CSS framework.
+	3. Completed page sitemap.
+
+For the time being, I am planning on developing with a Preact front-end, which is a highly optimized and minified variant of React. The end goal is to deliver a performant platform, which Preact will help in doing.
+
+#### Phase 2
+Phase 2 is the largest portion of the project since it involves:
+1. Back-end completion.
+	1. During Phase 2, the back-end must be completed up to spec in Express with completed and deployed endpoints in use by the front-end. In specific, the following metrics are being targeted for the time being, although more may be added as development progresses: 
+		1. Top level:
+			1. Block stats
+				1. Tipset Height
+				2. Number of blocks in tipset
+				3. Hash of parent block in tipset
+				4. Avg. block time
+				5. Last block miner address & PeerID
+				6. Storage Power
+				7. Number of messages in block
+				8. Block Size
+				9. Block Reward
+				10. Timestamp
+		2. Secondary:
+			1. All secondary metrics will be added, except for:
+				1. Average sector size
+				2. Actors:
+					1. Node latency
+2. Exposed API completion.
+	1. The exposed API will be a simple Express outgoing endpoint available at api.filexplorer.org which serves commonly used developer metrics. It will allow: 
+		1. GET requests for commonly used metrics (think: high-level deal data)
+		2. POST requests to retrieve messages, block information, transaction details.
+3. Filecoin node to MongoDB script completion.
+	1. This script will allow new transactions to be stored in a MongoDB cached instance allowing for faster retrieval, and better historic data retrieval. A large part of block explorers is their ability to index complete histories of blockchain metadata, and thus, this script will poll the node endpoint at quarter-second intervals, timestamping its requests.
+4. And, finally, Front-end completion:
+	1. By this point of time, most of the Preact components will have already have been built (with the framing in Phase 1). Thus, it is only a matter of adding them to their respective pages, setting up props, and using Axios to retrieve data from the back-end.
+		1. A large focus of the front-end will be maintaining responsiveness.
+		2. Victory and other data visualization libraries will be used to display network heatmaps, charts, and visualize other statistics.
+		3. A universal search bar (at the top) will be the main focal point of the front-end which will allow for POST requests including block peerID, addresses, content ID, and hash as the message.
```

Comment:
If you're using MongoDB how will you support universal search? MongoDB requires that queries be pre-indexed ahead of time. Otherwise it can only do simple text search.